Check for DLC presence in one spot

luckysori opened the pull request:
Actually fixes what https://github.com/itchysats/itchysats/pull/2835 set out to fix.

Patch e39a0134950d92931a8534d7d8c3f4380d908e0c was a step in the right direction in terms of suppressing false positives about not being able to auto-rollover CFDs without a DLC, it wasn't enough.

It was insufficient because we were checking for DLC presence in more than one spot in the same method. By checking in only one spot we can simplify the method and also ensure that we check if the CFD is empty _before_ we check for DLC presence.

da-kami commented on model/src/cfd.rs:
```diff
@@ -704,16 +698,19 @@ impl Cfd {
-        let expiry_timestamp = self.expiry_timestamp().ok_or(CannotRollover::NoDlc)?;
+        self.can_rollover()?;
+
+        let dlc = match &self.dlc {
```
❓Any particular reason why you prefer the `match` over the `ok_or`? I think it should be equivalent.

luckysori replied:
Nope, I just forgot!

klochowicz commented on model/src/cfd.rs:
```diff
@@ -704,16 +698,19 @@ impl Cfd {
+        let dlc = match &self.dlc {
+            Some(dlc) => dlc,
+            None => return Err(CannotRollover::NoDlc),
+        };
```
perhaps neater:

```suggestion
        let dlc = self.dlc.as_ref().ok_or(CannotRollover::NoDlc)?;
```

luckysori replied:
Yep, for sure!

luckysori committed:
fix(auto-rollover): Check for DLC presence in one spot

Patch e39a0134950d92931a8534d7d8c3f4380d908e0c was a step in the right
direction in terms of suppressing false positives about not being able
to auto-rollover CFDs without a DLC, it wasn't enough.

It was insufficient because we were checking for DLC presence in more
than one spot in the same method. By checking in only one spot we can
simplify the method and also ensure that we check if the CFD is empty
_before_ we check for DLC presence.
